README.md

```diff
@@ -56,57 +57,55 @@ import open from 'open';
 import path from 'path';
 
 async function uploadWithTurbo() {
-  // load your JWK directly to authenticate
-  const arweave = Arweave.init({});
   const jwk = JSON.parse(fs.readFileSync('./my-jwk.json', 'utf-8'));
-  const address = await arweave.wallets.jwkToAddress(jwk);
-  const turbo = TurboFactory.authenticated({ privateKey: jwk });
-
-  // or provide your own signer
-  // const signer = new ArweaveSigner(jwk);
-  // const turbo = TurboFactory.authenticated({ signer });
-
-  // get the wallet balance
-  const { winc: balance } = await turbo.getBalance();
+  const signer = new ArweaveSigner(jwk);
+  const turbo = TurboFactory.authenticated({ signer });
 
-  // prep file for upload
-  const filePath = path.join(__dirname, './my-image.png');
-  const fileSize = fs.statSync(filePath).size;
-
-  // get the cost of uploading the file
-  const [{ winc: fileSizeCost }] = await turbo.getUploadCosts({
-    bytes: [fileSize],
-  });
-
-  // check if balance greater than upload cost, and if in browser
-  if (balance < fileSizeCost && window !== undefined) {
-    const { url } = await turbo.createCheckoutSession({
-      amount: fileSizeCost,
-      owner: address,
-      // add a promo code if you have one
-    });
-
-    // open the URL to top-up if in browser
-    window.open(url, '_blank');
-  } else {
-    // otherwise, print the URL to the console
-    console.log('Please top up your balance via the CLI before uploading', {
-      balance,
-      fileSizeCost,
+  try {
+    // upload some simple data - log upload progress events
+    const { id, owner, dataCaches, fastFinalityIndexes } = await turbo.upload({
+      data: 'Hello, world!',
+      events: {
+        onUploadProgress: ({ totalBytes, processedBytes }) => {
+          console.log('Upload progress:', { totalBytes, processedBytes });
+        },
+        onUploadError: ({ error }) => {
```

Comment:
Why does error need to be unpacked? Will check further down in the PR but my guess would be to leave room for more error-related data in the future? If so it might make more sense to have a custom error object with more fields for whatever we need. I think it's just unusual to have to unpack an error in this way.

Reply:
`onUploadError` does provide additional context `step: 'upload' | 'signing'` to indicate which step failed. but this could also be done via custom error contexts. 

we have other use for a custom `SigningError` - right now when signing fails on `uploadFile` or `uploadFolder` we just throw a generic error. 

in this case - `onUploadError` would just be an `FailedRequestError` and `onSigningError` would be a `SigningError`, removing the need for the desctructured type

Reply:
moved to providing just the `Error` in [c926e23](https://github.com/ardriveapp/turbo-sdk/pull/254/commits/c926e232df64893ea8f819e4ef90b4b47705902e)

this leaves room for us to add the custom errors later